static/app/components/inputGroup.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  createContext,
+  forwardRef,
+  useContext,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import styled from '@emotion/styled';
+
+import _TextArea from 'sentry/components/forms/controls/textarea';
+import _Input, {InputProps} from 'sentry/components/input';
+import space from 'sentry/styles/space';
+import {FormSize} from 'sentry/utils/theme';
+
+interface InputContext {
+  /**
+   * Props to be passed to `Input`. When `InputGroup` is used, it is
+   * recommended to add input props (`size`, `disabled`) to `InputGroup`
+   * rather than `Input`, so that other elements in the group
+   * (`InputLeadingItems`, `InputTrailingItems`) also know about them.
+   */
+  inputProps: Partial<InputProps>;
+  /**
+   * Width of the leading items wrap, to be added to `Input`'s padding.
+   */
+  leadingWidth?: number;
+  setLeadingWidth?: (width: number) => void;
+  setTrailingWidth?: (width: number) => void;
+  /**
+   * Width of the trailing items wrap, to be added to `Input`'s padding.
+   */
+  trailingWidth?: number;
+}
+export const InputGroupContext = createContext<InputContext>({inputProps: {}});
+
+/**
+ * Context provider for input group. To be used alongisde `Input`, `InputLeadingItems`,
+ * and `InputTrailingItems`:
+ *   <InputGroup>
+ *     <InputLeadingItems> … </InputLeadingItems>
+ *     <Input />
+ *     <InputTrailingItems> … </InputTrailingItems>
+ *   </InputGroup>
+ */
+export function InputGroup({children, ...inputProps}: InputProps) {
```

Comment:
I would actually rather this just not take these props at all, and it recieves them via context from the `Input` used inside of it?

Reply:
Hmm lemme try doing that. Was concerned it would result in some layout flashes when the whole thing mounts

Reply:
Yeah it may, not sure what the workaround will be 😬

Reply:
Okay I used `useLayoutEffect` and it doesn't seem to flash on mount! https://github.com/getsentry/sentry/pull/40270/commits/cf42979525ae28279efc96c427be3da063c68be2